<a href="https://colab.research.google.com/github/Isabel160609/Cuadernos-colab-Pyton/blob/main/Copia_de_Ejecicios_Estad%C3%ADstica_descriptiva.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Estadística Desciptiva**

**EJEMPLO**

Los pesos en kg de un grupo de personas son los siguientes:

[62, 76, 57, 74, 68, 83, 61, 87, 71, 81, 68, 77, 62, 74, 68, 68, 74, 66, 73, 84, 54, 72, 78, 69, 88, 63, 76, 59, 71, 66].

Calcula: la media, mediana, moda, el rango y la desviación típica.

In [ ]:
#Las librerías son:

import numpy as np
import statistics as stats

# Creamos el conjunto de datos(por ejemplo: pesos(kg))
pesos= [62, 76, 57, 74, 68, 83, 61, 87, 71, 81, 68, 77, 62, 74, 68, 68, 74, 66, 73, 84, 54, 72, 78, 69, 88, 63, 76, 59, 71, 66]

In [ ]:
# Estadísticas descriptivas
media = np.mean(pesos)  # Promedio
mediana = np.median(pesos)  # Valor central
moda = stats.mode(pesos)  # Valor más común
rango = np.max(pesos) - np.min(pesos)  # Diferencia entre el máximo y el mínimo
desviacion = np.std(pesos, ddof=1)  # Dispersión de los datos

# RESULTADOS DEL ANÁLISIS DESCRIPTIVO
print("-"*50, "\n")
print(f"***Análisis Estadístico Descriptivo***")
print("-"*50, "\n")

print(f"Media: {media}kg")
print(f"Mediana: {mediana}kg")
print(f"Moda: {moda}kg")
print(f"Rango: {rango}kg")
print(f"Desviación estándar: {desviacion:.2f}kg")

-------------------------------------------------- 

***Análisis Estadístico Descriptivo***
-------------------------------------------------- 

Media: 71.0kg
Mediana: 71.0kg
Moda: 68kg
Rango: 34kg
Desviación estándar: 8.65kg


**EJERCICIO 1:**

La edades en una sala de espera son:

edades = [20, 22, 22, 23, 25, 28, 30, 32, 65]

1. Calcule la media, la mediana, la moda, el rango y la desviación estándar.
2. ¿El conjunto tiene datos atípicos?

**¿El conjunto tiene datos atípicos?**

El 65 es un dato atipico ya que es muy diferente a los otros datos

In [ ]:
import numpy as np
import statistics as stats
edades = [20, 22, 22, 23, 25, 28, 30, 32, 65]

media = np.mean(edades)  # Promedio
mediana = np.median(edades)  # Valor central
moda = stats.mode(edades)  # Valor más común
rango = np.max(edades) - np.min(edades)  # Diferencia entre el máximo y el mínimo
desviacion = np.std(edades, ddof=1)  # Dispersión de los datos

# RESULTADOS DEL ANÁLISIS DESCRIPTIVO
print("-"*50, "\n")
print(f"***Análisis Estadístico Descriptivo***")
print("-"*50, "\n")

print(f"Media: {media:.2f} años")
print(f"Mediana: {mediana} años")
print(f"Moda: {moda} años")
print(f"Rango: {rango} años")
print(f"Desviación estándar: {desviacion:.2f} años")


-------------------------------------------------- 

***Análisis Estadístico Descriptivo***
-------------------------------------------------- 

Media: 29.666666666666668 años
Mediana: 25.0 años
Moda: 22 años
Rango: 45 años
Desviación estándar: 13.85 años


**EJERCICIO 2:** El Examen "Filtro".

Objetivo: Observar cómo dos valores altos (outliers) inflan el promedio.

Las notas de 15 alumnos son:

notas = [1, 2, 1.5, 2, 3, 2.5, 1, 2, 3, 2, 1, 1.5, 2, 10, 10]

Calcula:
1. Calcular media, mediana y la desviación estándar.
2. ¿Cuál representa mejor la realidad del curso?



2. **¿Cuál representa mejor la realidad del curso?**

En este caso, la **mediana** (2.00) representa mejor la realidad del curso.

*   La **media** (2.97) está inflada por los dos valores atípicos (10, 10), dando una impresión de que el rendimiento general del curso es más alto de lo que realmente es.
*   La **mediana**, al ser el valor central de los datos ordenados, es menos sensible a los valores extremos, lo que la convierte en una medida más robusta para describir la tendencia central de un conjunto de datos con outliers.

In [5]:
import numpy as np
import statistics as stats
notas = [1, 2, 1.5, 2, 3, 2.5, 1, 2, 3, 2, 1, 1.5, 2, 10, 10]
media = np.mean(notas)  # Promedio
mediana = np.median(notas)  # Valor central
moda = stats.mode(notas)  # Valor más común
rango = np.max(notas) - np.min(notas)  # Diferencia entre el máximo y el mínimo
desviacion = np.std(notas, ddof=1)  # Dispersión de los datos

print("-"*50, "\n")
print(f"***Análisis Estadístico Descriptivo***")
print("-"*50, "\n")
print(f"Media: {media:.2f}")
print(f"Mediana: {mediana:.2f}")
print(f"Moda: {moda}")
print(f"Rango: {rango:.2f}")
print(f"Desviación estándar: {desviacion:.2f}")

-------------------------------------------------- 

***Análisis Estadístico Descriptivo***
-------------------------------------------------- 

Media: 2.97
Mediana: 2.00
Moda: 2
Rango: 9.00
Desviación estándar: 2.92


**EJERCICIO 3:** El Club de los Abuelos (Sesgo Negativo)

Objetivo: Entender cómo valores pequeños arrastran la media hacia la izquierda.

Las edades en un centro de jubilados con 3 nietos de visita son:

edades = [70, 72, 75, 80, 82, 70, 75, 80, 85, 78, 10, 12, 11]

Calcula:
1. Calcular media, mediana y la desviación estándar.
2. Comprobar la regla: Media < Mediana < Moda.


In [6]:
import numpy as np
import statistics as stats
edades = [70, 72, 75, 80, 82, 70, 75, 80, 85, 78, 10, 12, 11]
media = np.mean(edades)  # Promedio
mediana = np.median(edades)  # Valor central
moda = stats.mode(edades)  # Valor más común
rango = np.max(edades) - np.min(edades)  # Diferencia entre el máximo y el mínimo
desviacion = np.std(edades, ddof=1)  # Dispersión de los datos
print("-"*50, "\n")
print(f"***Análisis Estadístico Descriptivo***")
print("-"*50, "\n")
print(f"Media: {media:.2f}")
print(f"Mediana: {mediana:.2f}")
print(f"Moda: {moda}")
print(f"Rango: {rango:.2f}")
print(f"Desviación estándar: {desviacion:.2f}")
print(f"la regla de Media < Mediana < Moda es: media = {media:.2f}, mediana = {mediana:.2f}, moda = {moda}")

-------------------------------------------------- 

***Análisis Estadístico Descriptivo***
-------------------------------------------------- 

Media: 61.54
Mediana: 75.00
Moda: 70
Rango: 75.00
Desviación estándar: 29.16
 la regla de Media < Mediana < Moda es: media = 61.54, mediana = 75.00, moda = 70


2. **Comprobar la regla: Media < Mediana < Moda.**

La regla `Media < Mediana < Moda` no se cumple en este caso. Los valores calculados son:
*   **Media:** 61.54
*   **Mediana:** 75.00
*   **Moda:** 70

Si bien la Media (61.54) es menor que la Mediana (75.00), la Mediana (75.00) no es menor que la Moda (70). En este conjunto de datos, se observa que la Mediana es mayor que la Moda (75.00 > 70).

Esto ocurre debido a la presencia de los `outliers` (10, 11, 12) que "arrastran" la media significativamente hacia valores más bajos. Aunque la distribución general del grupo principal de edades (70-85) es más alta, estos valores atípicos distorsionan la relación esperada entre la media, mediana y moda para una distribución sesgada 'negativamente' o 'hacia la izquierda' ideal. La moda se mantiene en 70, que es un valor frecuente dentro del grupo de edades mayores, mientras que la mediana se sitúa en 75, también en el grupo de edades mayores, pero la media se ve fuertemente afectada por los valores pequeños.

In [7]:
import numpy as np
import statistics as stats
edades = [70, 72, 75, 80, 82, 70, 75, 80, 85, 78]
media = np.mean(edades)  # Promedio
mediana = np.median(edades)  # Valor central
moda = stats.mode(edades)  # Valor más común
rango = np.max(edades) - np.min(edades)  # Diferencia entre el máximo y el mínimo
desviacion = np.std(edades, ddof=1)  # Dispersión de los datos
print("-"*50, "\n")
print(f"***Análisis Estadístico Descriptivo***")
print("-"*50, "\n")
print(f"Media: {media:.2f}")
print(f"Mediana: {mediana:.2f}")
print(f"Moda: {moda}")
print(f"Rango: {rango:.2f}")
print(f"Desviación estándar: {desviacion:.2f}")
print(f"la regla de Media < Mediana < Moda es: media = {media:.2f}, mediana = {mediana:.2f}, moda = {moda}")

-------------------------------------------------- 

***Análisis Estadístico Descriptivo***
-------------------------------------------------- 

Media: 76.70
Mediana: 76.50
Moda: 70
Rango: 15.00
Desviación estándar: 5.14
la regla de Media < Mediana < Moda es: media = 76.70, mediana = 76.50, moda = 70


3. **¿Por qué la regla Media < Mediana < Moda no se cumple sin los valores atípicos?**

Incluso después de quitar los valores atípicos (10, 11, 12), la regla `Media < Mediana < Moda` no se cumple. Los nuevos valores son:
*   **Media:** 76.70
*   **Mediana:** 76.50
*   **Moda:** 70

En este caso, la **Media (76.70)** es ligeramente **mayor que la Mediana (76.50)**, y la **Moda (70)** es el valor **más bajo** de los tres. Esta configuración no corresponde a una distribución sesgada negativamente (donde la Media sería menor que la Mediana y la Mediana menor que la Moda).

En cambio, la relación observada (Media ≈ Mediana, con la Moda siendo el valor más bajo) sugiere que la distribución es más **cercana a ser simétrica** o ligeramente **sesgada positivamente** (hacia la derecha), donde la Moda < Mediana < Media. El hecho de que la Moda sea 70 (uno de los valores más bajos en el conjunto sin outliers) y la Media sea ligeramente superior a la Mediana, indica que no hay una cola de valores bajos 'arrastrando' la media como se esperaría en un sesgo negativo.

**Si quieres que tu pizza llegue siempre a la misma hora, ¿cuál eliges?**

Elegiria la segunda por que tiene una desviacion estandar menor

**EJERCICIO 4:** El Desafío de las Pizzerías (Comparación de Sesgo)

Objetivo: Comparar dos conjuntos de datos y decidir cuál es más confiable.

Tiempos de entrega en minutos:

pizzeria_A = [20, 22, 21, 19, 20, 25, 80, 90]
pizzeria_B = [35, 36, 34, 35, 37, 35, 34, 36]

Calcula:
1. Calcular la Desviación Estándar (np.std(datos, ddof=1)).
2. Si quieres que tu pizza llegue siempre a la misma hora, ¿cuál eliges?

**Si quieres que tu pizza llegue siempre a la misma hora, ¿cuál eliges?**

Elegiria la segunda por que tiene una desviacion estandar menor

In [8]:
import numpy as np
import statistics as stats
pizzeria_A = [20, 22, 21, 19, 20, 25, 80, 90]
pizzeria_B = [35, 36, 34, 35, 37, 35, 34, 36]
desviacionA = np.std(pizzeria_A, ddof=1)
desviacionB = np.std(pizzeria_B, ddof=1)
print(f"Desviación estándar: {desviacionA:.2f}")
print(f"Desviación estándar: {desviacionB:.2f}")

Desviación estándar: 29.72
Desviación estándar: 1.04
